dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/TextStreamingResult.cs

```diff
@@ -12,29 +13,47 @@ namespace Microsoft.SemanticKernel.Connectors.AI.OpenAI.AzureSdk;
 
 internal sealed class TextStreamingResult : ITextStreamingResult, ITextResult
 {
-    private readonly StreamingChoice _choice;
-
     public ModelResult ModelResult { get; }
 
-    public TextStreamingResult(StreamingCompletions resultData, StreamingChoice choice)
+    public TextStreamingResult(Completions completions, IReadOnlyList<Choice> choices)
     {
-        this.ModelResult = new ModelResult(resultData);
-        this._choice = choice;
+        this.ModelResult = new ModelResult(completions);
+        this._choices = choices;
     }
 
     public async Task<string> GetCompletionAsync(CancellationToken cancellationToken = default)
     {
         var fullMessage = new StringBuilder();
-        await foreach (var message in this._choice.GetTextStreaming(cancellationToken).ConfigureAwait(false))
+        await foreach (var message in this.GetCompletionStreamingAsync(cancellationToken).ConfigureAwait(false))
         {
             fullMessage.Append(message);
         }
 
         return fullMessage.ToString();
     }
 
-    public IAsyncEnumerable<string> GetCompletionStreamingAsync(CancellationToken cancellationToken = default)
+    public async IAsyncEnumerable<string> GetCompletionStreamingAsync([EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        int i = 0;
+        while (i < this._choices.Count)
+        {
+            yield return this._choices[i].Text;
+
+            i++;
+
+            // Wait for next choice update...
+            while (!this._isStreamEnded && i >= this._choices.Count)
+            {
+                await Task.Delay(50, cancellationToken).ConfigureAwait(false);
+            }
+        }
+    }
```

Comment:
Same concerns as earlier.  This producer/consumer implementation should be done differently.